qiskit/optimization/problems/quadratic_program.py

```diff
@@ -851,6 +853,216 @@ def substitute_variables(
         """
         return SubstituteVariables().substitute_variables(self, constants, variables)
 
+    def to_ising(self) -> Tuple[OperatorBase, float]:
+        """Return the Ising Hamiltonian of this problem.
+
+        Returns:
+            The qubit operator of the problem and the shift value.
+
+        Raises:
+            QiskitOptimizationError: If a variable type is not binary.
+            QiskitOptimizationError: If constraints exist in the problem.
+        """
+        # if problem has variables that are not binary, raise an error
+        if self.get_num_vars() > self.get_num_binary_vars():
+            raise QiskitOptimizationError('The type of variable must be a binary variable. '
+                                          'Use a QuadraticProgramToQubo converter to convert '
+                                          'integer variables to binary variables. '
+                                          'If the problem contains continuous variables, '
+                                          'currently we can not apply VQE/QAOA directly. '
+                                          'you might want to use an ADMM optimizer '
+                                          'for the problem. ')
+
+        # if constraints exist, raise an error
+        if self.linear_constraints \
+                or self.quadratic_constraints:
+            raise QiskitOptimizationError('An constraint exists. '
+                                          'The method supports only model with no constraints. '
+                                          'Use a QuadraticProgramToQubo converter. '
+                                          'It converts inequality constraints to equality '
+                                          'constraints, and then, it converters equality '
+                                          'constraints to penalty terms of the object function.')
+
+        # initialize Hamiltonian.
+        num_nodes = self.get_num_vars()
+        pauli_list = []
+        shift = 0
```

Comment:
In `from_ising` this variable is called `offset`, could you give them both the same name? (Either shift or offset)